sn_interface/src/network_knowledge/mod.rs

```diff
@@ -690,4 +696,52 @@ mod tests {
             assert_eq!(supermajority(n + 2), 2 * a + 2);
         }
     }
+
+    #[test]
+    fn signed_sap_field_should_not_be_changed_if_the_update_is_for_a_different_prefix() -> Result<()>
+    {
+        let mut rng = thread_rng();
+        let sk_gen = SecretKeySet::random(0, &mut rng);
+        let peer = Peer::new(XorName::random(&mut rng), gen_addr());
+        let (mut knowledge, _) = NetworkKnowledge::first_node(peer, sk_gen.clone())?;
+
+        // section 1
+        let (sap1, _, sk_1) = random_sap(prefix("1")?, 0, 0, None);
+        let sap1 = section_signed(&sk_1.secret_key(), sap1)?;
+        let our_node_name_prefix_1 = sap1.prefix().name();
+        let section_tree_update = {
+            let sig = bincode::serialize(&sap1.section_key())
+                .map(|bytes| sk_gen.secret_key().sign(&bytes))?;
+            let mut proof_chain = knowledge.section_chain();
+            proof_chain.insert(&sk_gen.secret_key().public_key(), sap1.section_key(), sig)?;
+            SectionTreeUpdate::new(sap1.clone(), proof_chain)
+        };
+        assert!(knowledge.update_knowledge_if_valid(
+            section_tree_update,
+            None,
+            &our_node_name_prefix_1
+        )?);
+        assert_eq!(knowledge.signed_sap, sap1);
+
+        // section with different prefix (0) and our node name doesn't match
+        let (sap0, _, sk_0) = random_sap(prefix("0")?, 0, 0, None);
+        let sap0 = section_signed(&sk_0.secret_key(), sap0)?;
+        let section_tree_update = {
+            let sig = bincode::serialize(&sap0.section_key())
+                .map(|bytes| sk_gen.secret_key().sign(&bytes))?;
+            let mut proof_chain = SectionsDAG::new(sk_gen.secret_key().public_key());
+            proof_chain.insert(&sk_gen.secret_key().public_key(), sap0.section_key(), sig)?;
+            SectionTreeUpdate::new(sap0, proof_chain)
```

Comment:
Consider using `section_signed` here instead of `bincode::serialize | sign`.
```suggestion
            let signed_key = section_signed(sk_gen.secret_key(), sap0.section_key());
            let mut proof_chain = SectionsDAG::new(sk_gen.secret_key().public_key());
            proof_chain.insert(&sk_gen.secret_key().public_key(), signed_key.value, signed_key.sig.signature)?;
            SectionTreeUpdate::new(sap0, proof_chain)
```

Reply:
Better yet, use your new `gen_section_tree_update()` helper

Reply:
Yeah! Totally forgot about it! I've made the changes, also updated a few other test cases to use it.